openhtf/io/frontend/__main__.py

```diff
@@ -32,97 +32,207 @@
-@app.route('/station/<station_id>/template/<template>/')
-def station_template(station_id, template):
-  """Handle requests for individual pieces of station pages."""
-  state = state = query_framework(station_id)
-  return flask.render_template(
-      '%s.html' % template, name=station_id, state=state)
-
-
-@app.route('/station/<name>/prompt/', methods=['GET'])
-def get_prompt(name):
-  """Handle a request for a station's prompt state."""
-  state = query_framework(name)
-  print('HACK 1')
-  if ('prompt' not in state) or (state['prompt'] == 'None'):
-    return 'NO_PROMPT'
-  return state['prompt']['id']
-
-
-@app.route('/station/<name>/prompt/<id>/', methods=['POST'])
-def prompt(name, id):
-  """Handle requests that respond to prompts."""
-  msg = json.JSONEncoder().encode({'id': id, 'response': flask.request.data})
-  query_framework(name, method='POST', message=msg)
-  return 'OK'
+gflags.DEFINE_integer('discovery_interval_s', 5,
+                      'Seconds to wait between station discovery attempts.')
+gflags.DEFINE_string('discovery_address',
+                     multicast.DEFAULT_ADDRESS,
+                     'Multicast address to ping for station discovery.')
+gflags.DEFINE_integer('discovery_port',
+                      multicast.DEFAULT_PORT,
+                      'Multicast port to use for station discovery.')
+gflags.DEFINE_integer('discovery_ttl',
+                      multicast.DEFAULT_TTL,
+                      'TTL for station discovery pings.')
+
+
+_LOG = logging.getLogger(__name__)
+
+UNKNOWN_STATION_ID = 'UNKNOWN_STATION'
+
+
+class Station(mutablerecords.Record(  # pylint: disable=too-few-public-methods,no-init
+    'Station', [], {'station_id': UNKNOWN_STATION_ID, 'state': None})):
+  """Represents a station seen on the local network.
+
+  Attributes:
+    station_id: Station's ID string.
+    state: A dictionary representation of station state with two keys:
+      framework: A dictionary representation of a TestExecutor object.
+      test: A dictionary representation of a TestState object.
+    """
+
+
+class StationStore(threads.KillableThread):
+  """Self-updating store of stations visible on the local network(s).
+
+  Station data is stored in the 'stations' attribute, a dictionary mapping
+  tuples of (host_ip, port) to Station records.
+  """
+  def __init__(self):
+    super(StationStore, self).__init__()
+    self._live = False
+    self.hostname = socket.gethostname()
+    self.stations = {}
+
+  @staticmethod
+  def MessageFramework(host, port, method='GET', message=None):
+    """Interact with a running instance of openhtf via its HTTP API.
+
+    Args:
+      host: The IP address of the host running openhtf.
+      port: The TCP port of openhtf's HTTP API on host.
+      method: 'GET' or 'POST'.
+      message: Prompt response for POSTs. Ignored for GETs.
+    """
+    result = None
+    conn = httplib.HTTPConnection(host, int(port))
+    args = [method, '/']
+    if message and method == 'POST':
```

Comment:
maybe log something that 'message' is getting ignored if it's passed but the method isn't POST